archeryutils/targets.py

```diff
@@ -60,41 +63,32 @@ def __init__(
                 f"""Please select from '{"', '".join(systems)}'."""
             )
 
-        if native_dist_unit in (
-            "Yard",
-            "yard",
-            "Yards",
-            "yards",
-            "Y",
-            "y",
-            "Yd",
-            "yd",
-            "Yds",
-            "yds",
-        ):
+        if native_dist_unit in DistanceUnits.yard:
             native_dist_unit = "yard"
-        elif native_dist_unit in (
-            "Metre",
-            "metre",
-            "Metres",
-            "metres",
-            "M",
-            "m",
-            "Ms",
-            "ms",
-        ):
+            distance *= YARD_TO_METRE
+        elif native_dist_unit in DistanceUnits.metre:
             native_dist_unit = "metre"
         else:
             raise ValueError(
                 f"Distance unit '{native_dist_unit}' not recognised. "
                 "Select from 'yard' or 'metre'."
             )
 
-        self.diameter = diameter
+        if diameter_unit in DistanceUnits.cm:
+            diameter_unit = "cm"
+            diameter *= CM_TO_METRE
+        elif diameter_unit in DistanceUnits.metre:
+            diameter_unit = "metre"
+        else:
+            raise ValueError(
+                f"Diamter unit '{diameter_unit} not recognised. "
+                "Select from 'cm' or 'metre'"
+            )
+
         self.native_dist_unit = native_dist_unit
-        self.distance = (
-            distance * YARD_TO_METRE if self.native_dist_unit == "yard" else distance
-        )
+        self.distance = distance
+        self.diameter_unit = diameter_unit
```

Comment:
Implemented but it gets quite verbose in the constructor:
Also a little confusing with the diameter being the first positional argument and then the units coming at the end, when writing out you would think to specify them together.

`Target(16, "Worcester", 20, "yards", indoor=True, native_diameter_unit="inches")`

Perhaps out of scope but I think for the same character count it would be better to have everything keyword and terser:

`Target(diam=16, diam_unit="inches", dist=20, dist_unit="yards", indoor=True)`

Or batch together lengths and units into a single object and pass those around instead, though thats more disruptive to the api:
```
#example length object, could implement this properly with typing etc
Length = namedtuple("Length", ["value", "unit"])
Target(diameter=Length(16, "inches"), distance=Length(20, "yards"), indoor=True)
```

Reply:
Mmmm yes - I would support removing `native_` from both as somewhat redundant now.
And I agree - when reading through your original changes I also felt it would be better to specify magnitude and unit together.
Batching together I don't think is ideal, as we want the units to be optional to simplify construction in general.

After some thought I think ordering arguments as `(scoring system, dia, dist, dia_unit, dist_unit...)` would be OK, as users can construct simply using assumed units, but if specifying as kwargs could group the unit with the distance as they please e.g. `("10_zone", dia=16, dia_unit="inches", dist=20, dist_unit="yards", indoor=True))` (I'd just need to check that definitely works). What do you think?

Reply:
Definitely agree with scoring system as first arg. 
Positional argument calls and relying on defaults can still be a bit fugly:

```
#not obvious which is distance and which is diameter!
Target("10_zone", 20, 40)
```

Reply:
What about batching but allowing either a single integer value with default units, or the tuple with units specified, and then dealing with it in the constructor? Then can have distance and diameter as keywords and leave the system as the first and only positional parameter

```
#implicit, default units
Target("10_zone", distance=90, diameter=122)
 
#explicit units, can mix and match with default units, order of distance vs diameter doesn't matter, still fairly concise
Target("10_zone", distance=(40, "metres"), diameter=40)
Target("Worcster", diameter=(16, "inches"), distance=(20, "yards"), indoors=True)
```

Reply:
After thinking about this I think this sort of thing is probably the best approach.
With a professional hat on I'd probably say to implement in an alternative PR, but happy for it to be done here if you feel up to it.

Reply:
Such a change will also require updating the loader (and other things...?) which I am happy to look at if you want.

Reply:
I'm happy to keep it seperate, this one already expanded in scope quite a bit.

Reply:
Yep, OK, I'll open an issue and do it once this PR is merged.